acme/order.go

```diff
@@ -111,17 +115,81 @@ func newOrder(db nosql.DB, ops OrderOptions) (*order, error) {
 		return nil, err
 	}
 
-	// Update the "order IDs by account ID" index //
-	oids, err := getOrderIDsByAccount(db, ops.AccountID)
+	var oidHelper = orderIDsByAccount{}
+	_, err = oidHelper.addOrderID(db, ops.AccountID, o.ID)
+	if err != nil {
+		return nil, err
+	}
+	return o, nil
+}
+
+type orderIDsByAccount struct{}
+
+// addOrderID adds an order ID to a users index of in progress order IDs.
+// This method will also cull any orders that are no longer in the `pending`
+// state from the index before returning it.
+func (oiba orderIDsByAccount) addOrderID(db nosql.DB, accID string, oid string) ([]string, error) {
```

Comment:
It's curious the use of an empty type for these two methods. Any reason for this?

Reply:
I figured these would be methods on the authDB once I make that change. This is sort of staging that.